fe/fe-core/src/main/java/com/starrocks/sql/ast/ColumnDef.java

```diff
@@ -266,9 +267,28 @@ public Type getType() {
         return typeDef.getType();
     }
 
+    public void analyze(boolean isOlap, boolean isInternalCatalog, String engineName) throws AnalysisException {
+        if (isInternalCatalog) {
+            if (!isOlap && !isAllowNull) {
+                // prevent not null for external table
+                if (engineName.equalsIgnoreCase(EngineType.HIVE.name()) ||
```

Comment:
maybe you could consider using `ConnectorType.isSupport()` or refactor the code accordingly.